x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryTranslatorTests.java

```diff
@@ -779,6 +779,85 @@ public void testAllCountVariantsWithHavingGenerateCorrectAggregations() {
                         + "\"gap_policy\":\"skip\"}}}}}"));
     }
 
+    public void testGroupSelection() {
```

Comment:
It would be great to split this up to more methods with more descriptive names and add more tests so that all 3 cases (no alias, alias, numeric reference) are covered, similarly to the integration tests in my comment: https://github.com/elastic/elasticsearch/pull/43072#pullrequestreview-267772000
Of course some of the cases, as mentioned, should be skipped as they won't work even with this fix.